### ISSUE TITLE
Properties of type `object` cause odd mapping behavior

### ISSUE THREAD

**Issue by GothikX** [agileobjects/AgileMapper#13]
Continuing on from #12, I figured out that a property of type object on one of my mapping targets (which wasn't mapped in any way yet) was causing the problem. Knowing this I can probably provide a more minimal example so I'll come back with details in a few minutes, if necessary.

**Comment by GothikX** [comment]
The class structure

```
    public class Source
    {
        public string Value { get; set; }
    }

    public class Target
    {
        public string Value { get; set; }

        public object Additional { get; set; }
    }
```

generates the following plan, which is very similar to what I was seeing:
```
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
// Map Source -> Target
// Rule Set: CreateNew
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 

sToTData =>
{
    try
    {
        var target = new Target();
        target.Value = sToTData.Source.Value;
        target.Additional = sToTData.Map(
            sToTData.Source,
            target.Additional,
            "Additional",
            0);

        return target;
    }
    catch (Exception ex)
    {
        throw MappingException.For(sToTData, ex);
    }
}

// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
// Map Source -> Source
// Rule Set: CreateNew
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 

sToSData =>
{
    try
    {
        var source = sToSData.Target ?? new Source();
        // No data source for Value

        return source;
    }
    catch (Exception ex)
    {
        throw MappingException.For(sToSData, ex);
    }
}
```

This plan doesn't throw any exceptions of course, mine did because of the additional mapped objects.

**Comment by SteveWilkes** [comment]
I've [fixed this](https://github.com/agileobjects/AgileMapper/commit/3a64a011e9213981b67ee5ad81c75ab523cdd45c) by ignoring object-typed members if there's no matching source member - is that the behaviour you'd expect?

I'm going to fix issue #10 as well, then push out a minor release tomorrow. Thanks again for the feedback :)

**Comment by GothikX** [comment]
Yeah I think that's fine. The library feels excellent by the way, really like the API and how easy it was to do exactly what I wanted :)

**Comment by SteveWilkes** [comment]
Thanks! Really good to hear :)

This should now be fixed as of [v0.10.2](https://www.nuget.org/packages/AgileObjects.AgileMapper).
